app/src/main/java/com/company/rentafield/di/NetworkModule.kt

```diff
@@ -76,66 +78,81 @@ object NetworkModule {
         }.build()
     }
 
-    @Provides
-    @Singleton
-    fun provideService(okHttpClient: OkHttpClient): RetrofitService {
+    private inline fun <reified T> provideRetrofitService(
+        okHttpClient: OkHttpClient,
+        baseUrl: String
+    ): T {
         return Retrofit.Builder()
-            .baseUrl(BASE_URL)
+            .baseUrl(baseUrl)
             .client(okHttpClient)
             .addConverterFactory(Json.asConverterFactory("application/json".toMediaType()))
             .build()
-            .create(RetrofitService::class.java)
+            .create(T::class.java)
     }
 
     @Provides
     @Singleton
     fun provideAiService(okHttpClient: OkHttpClient): RetrofitAiService {
-        return Retrofit.Builder()
-            .baseUrl(AI_URL)
-            .client(okHttpClient)
-            .addConverterFactory(Json.asConverterFactory("application/json".toMediaType()))
-            .build()
-            .create(RetrofitAiService::class.java)
+        return provideRetrofitService(okHttpClient, AI_URL)
+    }
+
+    @Provides
+    @Singleton
+    fun provideRetrofitAuthService(okHttpClient: OkHttpClient): RetrofitAuthService {
+        return provideRetrofitService(okHttpClient, BASE_URL)
     }
 
+    @Provides
+    @Singleton
+    fun provideRetrofitUserService(okHttpClient: OkHttpClient): RetrofitUserService {
+        return provideRetrofitService(okHttpClient, BASE_URL)
+    }
+
+    @Provides
+    @Singleton
+    fun provideRetrofitPlaygroundService(okHttpClient: OkHttpClient): RetrofitPlaygroundService {
+        return provideRetrofitService(okHttpClient, BASE_URL)
+    }
+
+
     @Provides
     @Singleton
     fun provideRemoteUserAuthorization(
-        retrofitService: RetrofitService
-    ): RemoteUserAuthorization = RemoteUserAuthorizationImpl(retrofitService)
+        retrofitAuthService: RetrofitAuthService
+    ): RemoteUserAuthorization = RemoteUserAuthorizationImpl(retrofitAuthService)
 
     @Provides
     @Singleton
     fun provideRemoteUserRepository(
-        retrofitService: RetrofitService
-    ): RemoteUserRepository = RemoteUserRepositoryImpl(retrofitService)
+        retrofitUserService: RetrofitUserService
+    ): RemoteUserRepository = RemoteUserRepositoryImpl(retrofitUserService)
 
     @Provides
     @Singleton
     fun provideRemoteUserBooking(
```

Comment:
I think we should use @Bind to bind the interface with the impl. please check that [link](https://developer.android.com/training/dependency-injection/hilt-android#inject-interfaces)